prov/efa/src/rxr/rxr_msg.c

```diff
@@ -993,40 +969,59 @@ ssize_t rxr_msg_trecv(struct fid_ep *ep_fid, void *buf, size_t len, void *desc,
 		      fi_addr_t src_addr, uint64_t tag, uint64_t ignore,
 		      void *context)
 {
+	struct fi_msg msg;
 	struct iovec msg_iov;
 
 	msg_iov.iov_base = (void *)buf;
 	msg_iov.iov_len = len;
 
-	return rxr_msg_generic_recv(ep_fid, &msg_iov, 1, src_addr, tag, ignore,
-				    context, ofi_op_tagged, 0);
+	msg.msg_iov = &msg_iov;
+	msg.iov_count = 1;
+	msg.addr = src_addr;
+	msg.context = context;
+	msg.desc = &desc;
+
+	return rxr_msg_generic_recv(ep_fid, &msg, tag, ignore, ofi_op_tagged, 0);
 }
 
 static
 ssize_t rxr_msg_trecvv(struct fid_ep *ep_fid, const struct iovec *iov,
 		       void **desc, size_t count, fi_addr_t src_addr,
 		       uint64_t tag, uint64_t ignore, void *context)
 {
-	return rxr_msg_generic_recv(ep_fid, iov, count, src_addr, tag, ignore,
-				    context, ofi_op_tagged, 0);
+	struct fi_msg msg;
+
+	msg.msg_iov = iov;
+	msg.iov_count = count;
+	msg.addr = src_addr;
+	msg.desc = desc;
+	msg.context = context;
+
+	return rxr_msg_generic_recv(ep_fid, &msg, tag, ignore, ofi_op_tagged, 0);
 }
 
 static
-ssize_t rxr_msg_trecvmsg(struct fid_ep *ep_fid, const struct fi_msg_tagged *msg,
+ssize_t rxr_msg_trecvmsg(struct fid_ep *ep_fid, const struct fi_msg_tagged *tagmsg,
 			 uint64_t flags)
 {
 	ssize_t ret;
+	struct fi_msg msg;
 
 	if (flags & FI_PEEK) {
-		ret = rxr_msg_peek_trecv(ep_fid, msg, flags);
+		ret = rxr_msg_peek_trecv(ep_fid, tagmsg, flags);
 		goto out;
 	} else if (flags & FI_CLAIM) {
-		ret = rxr_msg_claim_trecv(ep_fid, msg, flags);
+		ret = rxr_msg_claim_trecv(ep_fid, tagmsg, flags);
 		goto out;
 	}
 
-	ret = rxr_msg_generic_recv(ep_fid, msg->msg_iov, msg->iov_count, msg->addr,
-				   msg->tag, msg->ignore, msg->context,
+	msg.msg_iov = tagmsg->msg_iov;
```

Comment:
I'm probably missing something. Why are you doing this? Are there things from tagmsg that you don't want to pass along?

Reply:
The generic function serves any kind of recv, not just tagged ops, so it takes fi_msg as the argument, which is the lowest-common denominator of the support msg type structures.

Reply:
Ah alright